src/test/scala/is/hail/methods/LinearRegressionBurdenSuite.scala

```diff
@@ -0,0 +1,248 @@
+package is.hail.methods
+
+import is.hail.SparkSuite
+import is.hail.expr.TDouble
+import is.hail.keytable.KeyTable
+import is.hail.utils._
+import is.hail.variant.VariantDataset
+import org.apache.spark.sql.Row
+import org.testng.annotations.Test
+
+class LinearRegressionBurdenSuite extends SparkSuite {
+
+  /*
+  vdsBurden is shared by testWeightedSum, testWeightedSumWithImputation, and testMax.
+  Three genes overlap each other in the first 3 positions:
+
+              Position
+        1 2 3 4 5 6 7 8 9 10
+  Gene1 ---
+  Gene2   -
+  Gene3 -----
+  */
+
+  val vdsBurden: VariantDataset = hc.importVCF("src/test/resources/regressionLinear.vcf")
+    .annotateVariantsIntervals("src/test/resources/regressionLinear.interval_list", "va.genes", all=true)
+    .annotateVariantsExpr("va.weight = v.start.toDouble")
+    .annotateSamplesTable("src/test/resources/regressionLinear.pheno",
+      "Sample",
+      root = Some("sa.pheno"),
+      config = TextTableConfiguration(types = Map("Pheno" -> TDouble), missing = "0"))
+    .annotateSamplesTable("src/test/resources/regressionLinear.cov",
+      "Sample",
+      root = Some("sa.cov"),
+      config = TextTableConfiguration(types = Map("Cov1" -> TDouble, "Cov2" -> TDouble)))
+
+  def makeMap(sampleKT: KeyTable): Map[Any, IndexedSeq[java.lang.Double]] = sampleKT.collect().map { r =>
+    val s = r.asInstanceOf[Row].toSeq
+    s.head -> s.tail.map(_.asInstanceOf[java.lang.Double]).toIndexedSeq }
+    .toMap
```

Comment:
AFAICT, we always assume the output of this is a `Map[String, ...]`, if that's the case, we should assert that here (with a `.asInstanceOf[String]` and an appropriate return type).